### PR TITLE
Make maker more resilient against stream closures

### PR DESCRIPTION
**Pull request by klochowicz**
- reconnect price feed stream if it closes
- don't crash upon start if the orderbook stream is unavailable
- convey status of orderbook stream into maker health status

### CODE REVIEW

**Review by holzeis — APPROVED** [review]
LGTM 👍

File: maker/src/orderbook_ws.rs [hunk 1]
```diff
@@ -79,6 +86,7 @@ impl Client {
                 let authenticate = auth_fn;
                 match orderbook_client::subscribe_with_authentication(url, authenticate).await {
                     Ok((mut sink, mut stream)) => {
+                        let _ = orderbook_status.send(ServiceStatus::Online);
```
Comment (luckysori) [inline maker/src/orderbook_ws.rs:89]: :upside_down_face: We could also send this once we are authenticated. There's a relevant message handled below.

**Review by luckysori — APPROVED** [review]

File: maker/src/orderbook_ws.rs (hunk 1 above)
Comment (klochowicz) [reply maker/src/orderbook_ws.rs]: good idea! it looks perhaps a bit more convoluted now, but it's certainly more correct. :-)

**Comment by klochowicz** [comment]
@luckysori I've done the 2-step update especially for you ❤️‍🔥thank you for coming up with the idea, it's a good habit!

**Pull request merged**